internal/services/containers/container_registry_task_resource.go

```diff
@@ -0,0 +1,1573 @@
+	SourceType    string `tfschema:"source_type"`
+	RepositoryURL string `tfschema:"repository_url"`
+	Branch        string `tfschema:"branch"`
+	Auth          []Auth `tfschema:"auth"`
+}
+
+type SourceTrigger struct {
+	Name          string          `tfschema:"name"`
+	Enabled       bool            `tfschema:"enabled"`
+	Events        []string        `tfschema:"events"`
+	SourceSetting []SourceSetting `tfschema:"source_setting"`
+}
+
+type TimerTrigger struct {
+	Name     string `tfschema:"name"`
+	Enabled  bool   `tfschema:"enabled"`
+	Schedule string `tfschema:"schedule"`
+}
+
+type Identity struct {
+	Type        string   `tfschema:"type"`
+	IdentityIds []string `tfschema:"identity_ids"`
+	PrincipalId string   `tfschema:"principal_id"`
+	TenantId    string   `tfschema:"tenant_id"`
+}
+
+type RegistryCredential struct {
+	LoginServer string `tfschema:"login_server"`
+	UserName    string `tfschema:"username"`
+	Password    string `tfschema:"password"`
+	Identity    string `tfschema:"identity"`
+}
+
+type ContainerRegistryTaskModel struct {
+	Name                string                 `tfschema:"name"`
+	ContainerRegistryId string                 `tfschema:"container_registry_id"`
+	Identity            []Identity             `tfschema:"identity"`
+	AgentConfig         []AgentConfig          `tfschema:"agent_setting"`
+	AgentPoolName       string                 `tfschema:"agent_pool_name"`
+	IsSystemTask        bool                   `tfschema:"is_system_task"`
+	LogTemplate         string                 `tfschema:"log_template"`
+	Platform            []Platform             `tfschema:"platform_setting"`
+	Enabled             bool                   `tfschema:"enabled"`
+	TimeoutInSec        int                    `tfschema:"timeout_in_seconds"`
+	DockerStep          []DockerStep           `tfschema:"docker_step"`
+	FileTaskStep        []FileTaskStep         `tfschema:"file_task_step"`
+	EncodedTaskStep     []EncodedTaskStep      `tfschema:"encoded_task_step"`
+	BaseImageTrigger    []BaseImageTrigger     `tfschema:"base_image_trigger"`
+	SourceTrigger       []SourceTrigger        `tfschema:"source_trigger"`
+	TimerTrigger        []TimerTrigger         `tfschema:"timer_trigger"`
+	RegistryCredential  []RegistryCredential   `tfschema:"registry_credential"`
+	Tags                map[string]interface{} `tfschema:"tags"`
+}
+
+func userDataStateFunc(v interface{}) string {
+	switch s := v.(type) {
+	case string:
+		return utils.Base64EncodeIfNot(s)
+	default:
+		return ""
+	}
+}
+
+func (r ContainerRegistryTaskResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ContainerRegistryTaskName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RegistryID,
+		},
+		"platform_setting": {
```

Comment:
settings seems redundant?
```suggestion
		"platform": {
```